src/ophys_etl/modules/segmentation/postprocess_utils/roi_time_correlation.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+from ophys_etl.modules.segmentation.postprocess_utils.roi_types import (
+    SegmentationROI)
+
+
+def sub_video_from_roi(roi: SegmentationROI,
+                       video_data: np.ndarray) -> np.ndarray:
+    """
+    Get a sub-video that is flattened in space corresponding
+    to the video data at the ROI
+
+    Parameters
+    ----------
+    roi: SegmentationROI
+
+    video_data: np.ndarray
+        Shape is (ntime, nrows, ncols)
+
+    Returns
+    -------
+    sub_video: np.ndarray
+        Shape is (ntime, npix) where npix is the number
+        of pixels marked True in the ROI
+    """
+
+    xmin = roi.x0
+    ymin = roi.y0
+    xmax = roi.x0+roi.width
+    ymax = roi.y0+roi.height
+
+    sub_video = video_data[:, ymin:ymax, xmin:xmax]
+
+    mask = roi.mask_matrix
+    sub_video = sub_video[:, mask].reshape(video_data.shape[0], -1)
+    return sub_video
+
+
+def correlate_sub_video(sub_video: np.ndarray,
+                        key_pixel: np.ndarray,
+                        filter_fraction: float = 0.2) -> np.ndarray:
+    """
+    Correlated all of the pixels in a sub_video against
+    a provided time series using only the brightest N
+    timesteps
+
+    Parameters
+    ----------
+    sub_video: np.ndarray
+        Shape is (ntime, npixels)
+
+    key_pixel: np.ndarray
+        Shape is (ntime,)
+        This is the time series against which to correlate
+        the pixels in sub_video
+
+    filter_fraction: float
+        Keep the brightest filter_fraction timesteps when doing
+        the correlation (this is reckoned by the flux values in
+        key_pixel)
+
+    Returns
+    -------
+    corr: np.ndarray
+        Shape is (npix,)
+        These are the correlation values of the pixels in
+        sub_video against key_pixel
+    """
+
+    npix = sub_video.shape[1]
+    discard = 1.0-filter_fraction
+    th = np.quantile(key_pixel, discard)
+    mask = (key_pixel >= th)
+    sub_video = sub_video[mask, :]
+    key_pixel = key_pixel[mask]
```

Comment:
this seems to be a new flavor of `filter_fraction` behavior, where you're not considering the activity of `sub_video`.
Intentional?

Reply:
It is intentional, mostly for engineering concerns. If I can select the timestep mask based on a single trace, then I can quickly calculate the correlation with every ROI using `np.dot`. If each pair of pixels implies a different mask, then I have to do a loop over pixels which will be much slower.

I suppose I could just use the average of sub-video to calculate the mask once for all pixels.

I wasn't terribly alarmed by this because, back when I experimented with adding pairwise correlation (i.e. different mask for each pixel pair) to the feature vector segmenter, I didn't get noticably different results than when I computed a single mask for all pixel pairs.

(I suspect we are about to talk about this face-to-face, so I will leave this conversation ambiguous until we decide what options we want to test)